frontend/src/metabase-lib/expressions/resolver.js

```diff
@@ -144,13 +144,19 @@ export function resolve(expression, type = "expression", fn = undefined) {
     if (!clause) {
       throw new ResolverError(t`Unknown function ${op}`, expression.node);
     }
-    const { displayName, args, multiple, hasOptions } = clause;
+    const { displayName, args, multiple, hasOptions, validator } = clause;
     if (!isCompatible(type, clause.type)) {
       throw new ResolverError(
         t`Expecting ${type} but found function ${displayName} returning ${clause.type}`,
         expression.node,
       );
     }
+    if (typeof validator === "function") {
```

Comment:
I think we could rely on correct types here and rewrite the code to the following to make it cleaner:
```
if (validator) {
  const validationError = validator(...operands);
  if (validationError) {
    throw new ResolverError(validationError, expression.node);
  }
}
```

WDYT?

Reply:
Much improved, thanks. My JS is antiquated.